build.py

```diff
@@ -1514,6 +1540,37 @@ def add_cpu_libs_to_linux_dockerfile(backends, target_machine):
     return df
 
 
+def change_default_python_version_rhel(version):
+    # Example version 3.12.1 --> 3.12
+    major_minor_version = ".".join(version.split(".")[:2])
+    df = """
+# RHEL image has several python versions. It's important
+# to set the correct version, otherwise, packages that are
+# pip installed will not be found during testing.
+ENV PYVER {}
+ENV PYTHONPATH /opt/python/v
+RUN ln -sf /opt/python/cp${{PYVER/./}}* ${{PYTHONPATH}}
+ENV PYBIN ${{PYTHONPATH}}/bin
+ENV PYTHON_BIN_PATH ${{PYBIN}}/python${{PYVER}}
+ENV PATH ${{PYBIN}}:${{PATH}}
+""".format(
+        major_minor_version
+    )
+    df += """
+# The python library version available for install via 'yum install python3.X-devel' does not
+# match the version of python inside the RHEL base container. This means that python packages
+# installed within the container will not be picked up by the python backend stub process pybind
+# bindings. It must instead must be installed via pyenv.
+RUN curl https://pyenv.run | bash
+ENV PATH="/root/.pyenv/bin:$PATH"
+RUN eval "$(pyenv init -)"
```

Comment:
Why we need this version switch? is it needed for test purpose?

Reply:
We want to be aligned with DLFW which is using Python v3.12. The base container comes installed with the python 3.12 interpreter, but we also need to python3.12 libraries (e.g., libpython3.12.so) to build the python  backend. We install the library we need using `pyenv`, which was suggested to me via the DLFW team.